File: app/src/main/java/fr/paug/androidmakers/util/ThemeUtils.java
package fr.paug.androidmakers.util;

import android.content.Context;
import android.util.TypedValue;

import fr.paug.androidmakers.R;

public final class ThemeUtils {

    private static final ThreadLocal<TypedValue> TYPED_VALUE = new ThreadLocal<TypedValue>() {
        @Override
        protected TypedValue initialValue() {
            return new TypedValue();
        }
    };

    private ThemeUtils() {
        // No instances
    }

    public static void ensureRuntimeTheme(Context context) {
        final TypedValue tv = TYPED_VALUE.get();
        context.getTheme().resolveAttribute(R.attr.runtimeTheme, tv, true);
        if (tv.resourceId <= 0) {
            throw new IllegalArgumentException("runtimeTheme not defined in the preview theme");
        }
        context.setTheme(tv.resourceId);
    }

}

File: app/src/main/java/fr/paug/androidmakers/util/ThemeUtils.kt
package fr.paug.androidmakers.util

import android.content.Context
import android.util.TypedValue
import fr.paug.androidmakers.R

object ThemeUtils {
  private val TYPED_VALUE: ThreadLocal<TypedValue> = object : ThreadLocal<TypedValue>() {
    override fun initialValue(): TypedValue {
      return TypedValue()
    }
  }

  fun ensureRuntimeTheme(context: Context) {
    val tv = TYPED_VALUE.get()
    context.theme.resolveAttribute(R.attr.runtimeTheme, tv, true)
    require(tv.resourceId > 0) { "runtimeTheme not defined in the preview theme" }
    context.setTheme(tv.resourceId)
  }
}